### ISSUE TITLE
Make data formats friendlier to external editors

### ISSUE THREAD

**Issue by kemenaran** [zladx/LADX-Disassembly#375]
External data editor (like level editors) may need to edit a data file, or even replace it completely. For this reason, data files should be easy to edit programmatically.

I think the project should favor simple assembly for data files (over complicated constants and macros). Richer details can be inserted as comments, which keeps the file readable while making it easier to edit.

For instances, the entities definitions for each room could move from:

```asm
Overworld00Entities::
  entity $6, $7, ENTITY_MINI_MOLDORM
  entity $2, $4, ENTITY_HEART_PIECE
  entity $2, $6, ENTITY_CROW
  entities_end

Overworld01Entities::
; snip…
```

to

```asm
Overworld00Entities::
  db $76, $23 ; ENTITY_MINI_MOLDORM, x:6, y:7 
  db $42, $35 ; ENTITY_HEART_PIECE, x:2, y:4
  db $62, $7A ; ENTITY_CROW, x:2, y:6
  db $FF      ; end of list

Overworld01Entities::
; snip…
```

**Comment by Rangi42** [comment]
In my experience with [Polished Map](https://github.com/Rangi42/polished-map), high-level macro data files are *easier* to programatically read and write than  raw hex `db`s with comments, especially if you want to keep the comments in sync with edited data. Like in your example, a `entities_end` line would be easier to parse than looking for `db $FF` (or equivalents `db $ff`, or `db 255`, or `db -1`). (You would at least *some* parsing, to extract hex/decimal/binary numbers and ignore comments, and at that point you may as well be recognizing custom macros like `entity`.)

**Comment by kemenaran** [comment]
Indeed. @daid is having a look at at least splitting some tables to external files, which seems a better approach.
